app/src/lib/stores/app-store.ts

```diff
@@ -1041,6 +1041,19 @@ export class AppStore extends TypedBaseStore<IAppState> {
           tip.branch.tip.sha,
           inferredBranch.tip.sha
         )
+
+        // In the case that the aheadBehindCache doesn't have the needed data, try to
+        // request it directly from AheadBehindUpdater. This usually happens on initial load
+        // before AheadBehindUpdater has run though all the branches.
+        if (
+          aheadBehindOfInferredBranch === null &&
```

Comment:
I think we can remove this guard (and the lines 1040-1043 above) since if we're checking `this.comparisonCache` in `executeTaskAsync`, that should return the same result as would be found in the `compareState.aheadBehindCache` in this scope. i think that would remove an extra cache check and simplify the logic path here.

(the `aheadBehindUpdater` emits updates whenever it adds to its cache, which then gets propagated to the repository state's compare state)

Reply:
Addressed in https://github.com/desktop/desktop/pull/9319/commits/f150e1fa067e930ca2a3c1c5b30e70f841e3b5a2 !